GetDKAN/dkan
# Issue #3839 - Disentangle harvest register and revert

## Steps to Recreate

- Register a harvest (h1) with a dataset with a given identifier
- Run the h1 harvest
- Confirm the dataset was created
- Register a 2nd harvest (h2) with a new identifier that contains a slightly updated dataset with the same identifier as the first harvest (e.g. The json file was renamed in a new version.)
- Run the h2 harvest
- Confirm the dataset was updated
- Deregister the old h1 harvest: ddev drush dkan:harvest:deregister h1
- Note that this deleted the dataset even though it was also connected to the h2 harvest

## QA Steps

- Register a harvest using a local json file: ddev drush dkan:harvest:register --identifier=h1 --extract-uri=https://mylocal.ddev.site/sites/default/files/h1.json
- Run the h1 harvest: ddev drush dkan:harvest:run h1
- Confirm the dataset node was created
- Register a 2nd harvest using a local json file with an updated dataset with the same identifier as the first harvest: ddev drush dkan:harvest:register --identifier=h2 --extract-uri=https://mylocal.ddev.site/sites/default/files/h2.json
- Run the h2 harvest: ddev drush dkan:harvest:run h2
- Confirm that the dataset node was updated
- Deregister the h1 harvest: ddev drush dkan:harvest:deregister h1
- Asks for confirmation with a warning "If you deregister a harvest with published datasets, you will not be able to bulk revert the datasets connected to this harvest."
- -- After confirming, returns "Successfully deregistered the h1 harvest."
- Confirm that the harvest was deregistered: ddev drush dkan:harvest:list. (h1 should not be listed)
- Confirm that the dataset original created by the h1 harvest still remains
